lib/puppet/provider/iis_powershell.rb

```diff
@@ -57,6 +58,50 @@ def self.ps_manager
     PuppetX::IIS::PowerShellManager.instance("#{command(:powershell)} #{PuppetX::IIS::PowerShellCommon.powershell_args.join(' ')}")
   end
 
+  def self.ps_major_version
+    if @ps_major_version.nil?
+      version = PuppetX::IIS::PowerShellVersion.version
+      @ps_major_version = version.split('.').first.to_i
+    end
+    @ps_major_version
+  end
+
+  def self.add_powershell2_support
+    # - PowerShell 2.0 does not support autoload of modules therefore we must explicitly add the WebAdministration module
+    # - Must change the current location to the be the IIS: provider
+    # - Add the ConvertTo-JSON command support
+    ["Import-Module WebAdministration -ErrorAction Stop\ncd iis:\n",ps_script_content('json_1.7', @resource)]
```

Comment:
I meant it's a lot in one line, I didn't see the IIS provider part at first and it's hard to read.